### ISSUE TITLE
ALPN policy annotation requires TLS target groups

### ISSUE THREAD

**Issue by iAnomaly** [kubernetes-sigs/aws-load-balancer-controller#2145]
**Describe the bug**
ALPN policy annotation currently requires TLS target groups. This is unnecessary and prevents TLS terminated at NLB using a ACM certificate for example from setting ALPN policy via the annotation. I am using a TLS listener with a plain TCP backend/target group and I was able to manually set ALPN policy via the ELB console without issue. curl is also confirming the ALPN negotiation for HTTP2 is now working as expected after my change:

Before ALPN:
```
* Connected to ***.elb.us-east-1.amazonaws.com (***) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server did not agree to a protocol
```

After ALPN `HTTP2Preferred`:
```
* Connected to ***.elb.us-east-1.amazonaws.com (***) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use h2
```

**Steps to reproduce**
Configure a NLB with TLS listener and TCP/plaintext target group and apply the `service.beta.kubernetes.io/aws-load-balancer-alpn-policy` annotation. Nothing happens.

**Expected outcome**
`service.beta.kubernetes.io/aws-load-balancer-alpn-policy` should NOT check/require target group is using TLS.

**Environment**

* AWS Load Balancer controller version 2.2.1
* Kubernetes version 1.20.0
* Using EKS (yes/no), if so version? yes, 1.20.0

**Additional Context**:

**Comment by iAnomaly** [comment]
@julianxhokaxhiu also mentioned this issue in a comment in the original PR after it was merged: https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/1669#issuecomment-874770289

@M00nF1sh @kishorj Shall I submit a PR for this?

**Comment by M00nF1sh** [comment]
@iAnomaly Yes, feel free to summit a PR for this.

**Comment by kishorj** [comment]
@iAnomaly, if AWS allows the configuration, controller should not restrict it. Feel free to raise a PR.

**Comment by kishorj** [comment]
@iAnomaly, TLS target group was a requirement for ALPN configuration at some point in the past, hence the original limitation. I'm now able to create ALPN configuration without a TLS target group. I've merged your PR, thanks for the contribution.

**Comment by iAnomaly** [comment]
Closed by #2147

**Comment by julianxhokaxhiu** [comment]
Thank you for the fast fix, appreciated!
